### PR TITLE
List inactive projects on edit page

### PR DESCRIPTION
**Pull request by andreLumor**
Related to #67 
Created conditional that permits inactive projects to be listed, but only on edit page.

https://user-images.githubusercontent.com/36737050/153637379-738d098c-2b0b-49c4-8427-074785fdfe8c.mp4

### CODE REVIEW

File: app/admin/allocations.rb [hunk 1]
```diff
@@ -87,9 +87,13 @@
                                       .new(current_user.company)
                                       .not_allocated_including(f.object.user)
                                       .select(:id, :name)
-
+      
         input :user, as: :select, collection: company_users_not_allocated
-        input :project, collection: current_user.company.projects.active.order(:name)
+        if params[:action] == 'edit'
+          input :project, collection: current_user.company.projects.order(:name)
```
Comment (adbatista) [inline app/admin/allocations.rb:93]: why?

**Comment by wenderjean** [comment]
Let's discuss a better solution to this issue. https://github.com/Codeminer42/Punchclock/issues/67

File: app/admin/allocations.rb (hunk 1 above)
Comment (andreLumor) [reply app/admin/allocations.rb:93]: So you can't create a new allocation to a inactive project

File: app/admin/allocations.rb [hunk 2]
```diff
@@ -87,9 +87,9 @@
                                       .new(current_user.company)
                                       .not_allocated_including(f.object.user)
                                       .select(:id, :name)
-
+      
         input :user, as: :select, collection: company_users_not_allocated
-        input :project, collection: current_user.company.projects.active.order(:name)
+        input :project, collection: (current_user.company.projects.active.to_a | [@resource.project]).reject(&:blank?)
```
[the remark this answers is not in the document]
Comment (adbatista) [reply app/admin/allocations.rb:92]: still need to sort the results

File: app/admin/allocations.rb [hunk 3]
```diff
@@ -87,9 +87,9 @@
                                       .new(current_user.company)
                                       .not_allocated_including(f.object.user)
                                       .select(:id, :name)
-
+
```
Comment (wenderjean) [inline app/admin/allocations.rb:90]: Please, remove these trailing spaces here.

Comment (andreLumor) [reply app/admin/allocations.rb]: Done!

File: app/admin/allocations.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (andreLumor) [reply app/admin/allocations.rb]: Used sort_by(&:name).

File: app/admin/allocations.rb [hunk 4]
```diff
@@ -89,7 +89,7 @@
                                       .select(:id, :name)
 
         input :user, as: :select, collection: company_users_not_allocated
-        input :project, collection: current_user.company.projects.active.order(:name)
+        input :project, collection: (current_user.company.projects.active.to_a | [@resource.project]).reject(&:blank?).sort_by(&:name)
```
Comment (adbatista) [inline app/admin/allocations.rb:92]: can you add tests?

Comment (andreLumor) [reply app/admin/allocations.rb:92]: sure

Comment (andreLumor) [reply app/admin/allocations.rb:92]: should i create a allocations_spec.rb in spec/admin?
Or there is another place designed to test this issue?

File: spec/features/admin/allocations_spec.rb [hunk 5]
```diff
@@ -173,6 +177,12 @@
                         have_css(".row-start_at td", text: I18n.l(allocation.start_at, format: :long)) &
                         have_css(".row-end_at td", text: I18n.l(Date.today.next_month, format: :long))
       end
+
+      it "must show inactive project if user allocated" do
+        project.disable!
+        visit current_path
```
Comment (andreLumor) [inline spec/features/admin/allocations_spec.rb:183]: "visit current_path" is used to reload page after project disable.

File: spec/features/admin/allocations_spec.rb [hunk 6]
```diff
@@ -65,6 +65,10 @@
         expect(page).not_to have_text(project_not_active.name)
       end
 
+      it 'should not render nil project' do
```
Comment (andreLumor) [inline spec/features/admin/allocations_spec.rb:68]: Testing if page doesn't render blank project from resources.

**Pull request merged**